spec/classes/puppet_agent_windows_install_spec.rb

```diff
@@ -129,6 +129,38 @@
           end
         end
       end
+      context 'rubyplatform' do
+        facts = {
+          :architecture => 'x64',
+          :env_temp_variable => 'C:\tmp',
+          :kernelmajversion => kernelmajversion,
+          :osfamily => 'windows',
+          :puppetversion => '3.8.0',
+          :puppet_confdir => "#{values[:appdata]}/puppet/etc",
+          :mco_confdir => "#{values[:appdata]}/mcollective/etc",
+          :puppet_agent_pid => 42,
+          :system32 => 'C:\windows\sysnative',
+          :tmpdir => 'C:\tmp',
+        }
+        describe 'i386-ming32' do
+          let(:facts) { facts.merge({:rubyplatform => 'i386-ming32'}) }
+          it {
+            is_expected.to contain_exec('install_puppet.bat').with { {
+                             'command' => 'C:\windows\sysnative\cmd.exe /c start /b C:\windows\system32\cmd.exe /c "C:\tmp\install_puppet.bat"',
+                           } }
+
+          }
+        end
+        describe 'x86' do
+          let(:facts) { facts.merge({:rubyplatform => 'x86_64'}) }
+          it {
+            is_expected.to contain_exec('install_puppet.bat').with { {
+                             'command' => 'C:\windows\sysnative\cmd.exe /c start /b C:\windows\sysnative\cmd.exe /c "C:\tmp\install_puppet.bat"',
```

Comment:
If it passes, then it is still wrong.

The second call to cmd.exe is always `c:\Windows\system32\cmd.exe`. It's probably fine to just use that instead of a private variable.